### PR TITLE
SEC-490

### PR DESCRIPTION
**Pull request by wtcruft**
Addresses #490 !

### CODE REVIEW

File: app/controllers/concerns/lesson_planning.rb [hunk 1]
```diff
@@ -23,4 +24,14 @@ def current_planned_lesson(lesson)
       planned_lessons.
       find_or_initialize_by(lesson_id: lesson.id)
   end
+
+  def current_lesson_plan_missing_lesson_in_topic?(topic)
+    topic.lessons.each do |lesson|
```
Comment (wioux) [inline app/controllers/concerns/lesson_planning.rb:29]: Could also use `topic.lessons.any?{ ... }` here to be more declarative/functional. Non-blocking.

File: app/controllers/lesson_plans_controller.rb [hunk 2]
```diff
@@ -59,6 +70,22 @@ def create_lesson
     end
   end
 
+  def destroy_lessons_by_topic
+    topic = Topic.find(params[:topic_id])
+    @lesson_plan = current_lesson_plan
+    lessons = topic.lessons
+
+    lessons.each do |lesson|
+      @lesson_plan.planned_lessons.
```
Comment (wioux) [inline app/controllers/lesson_plans_controller.rb:79]: Non-blocking: `current_planned_lesson(lesson).destroy` should work here as well I think.

**Review by wioux — APPROVED** [review]
This looks great! Thanks for including tests. I left a few suggestions for simplifying a loop here and there, but they may be just nits/style preferences. I tested the branch locally and it worked :100:.

**Comment by wtcruft** [comment]
Thank you @wioux -- great suggestions! I've incorporated them, made sure the test files I edited pass, and ran through a bunch of manual click-tests. I'll wait for travis to finish but then it should be ready to go :)

**Pull request merged**